pkg/forwarder/forwarder.go

```diff
@@ -141,8 +142,13 @@ func (f *DefaultForwarder) retryTransactions(retryBefore time.Time) {
 
 	for _, t := range f.retryQueue {
 		if t.GetNextFlush().Before(retryBefore) {
-			f.waitingPipe <- t
-			transactionsExpvar.Add("Retried", 1)
+			select {
+			case f.lowPrio <- t:
+				transactionsExpvar.Add("Retried", 1)
+			default:
+				log.Errorf("workers retry queue is full, dropping transactions !")
```

Comment:
the message is a bit misleading since the `retry queue` here is different from the `retry queue` that's mentioned in the log message 10 lines below.

Instead of logging this, what do you think about logging one message at the end of this function that says something like:
```
Dropped <total> transactions in this retry attempt: <dropped_because_size_exceeded> transactions for exceeding the retry queue size limit, <dropped_because_workers_busy> transactions because the workers are too busy
```